pandas/core/indexes/base.py

```diff
@@ -197,13 +197,13 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None,
         # categorical
         if is_categorical_dtype(data) or is_categorical_dtype(dtype):
             from .category import CategoricalIndex
-            return CategoricalIndex(data, copy=copy, name=name, **kwargs)
+            return CategoricalIndex(data, dtype=dtype, copy=copy, name=name,
+                                    **kwargs)
 
         # interval
-        if is_interval_dtype(data):
+        if is_interval_dtype(data) or is_interval_dtype(dtype):
             from .interval import IntervalIndex
-            return IntervalIndex.from_intervals(data, name=name,
-                                                copy=copy)
+            return IntervalIndex(data, dtype=dtype, name=name, copy=copy)
```

Comment:
The `IntervalIndex` changes are more about future proofing; the `dtype` kwarg currently doesn't do anything for `IntervalIndex`, so there aren't really any tests that can be added.  I plan to create a PR to allow conversion between subtypes and initializing with non-inferred subtype (e.g. `interval[int64]` to `interval[float64]`) where I believe the same issue will come into play, so figured I'd put this code in place now while it's fresh on my mind.

Also, switched from `IntervalIndex.from_intervals` to the `IntervalIndex` constructor since the constructor does everything `from_intervals` does, but has more features and better testing.  Would eventually like to either remove `from_intervals` or just redirect it to the constructor.